lib/googleDrive/index.js

```diff
@@ -0,0 +1,185 @@
+import google from 'googleapis';
+import multiparty from 'multiparty';
+import googleAuth from './googleAuth';
+import {
+  FieldValidationError,
+} from '../tools';
+
+let drive;
+
+export function setup(googleApi = {
+  clientId: '',
+  clientSecret: '',
+  accessToken: '',
+  refreshToken: '',
+  expiryDate: '',
+}) {
+  drive = google.drive({ version: 'v3', auth: googleAuth(googleApi) });
+}
+
+/**
+ * You can extend it passing additional `fields` described
+ * here https://developers.google.com/drive/v3/reference/files#resource
+ * @typedef {object} GoogleDriveFileResource
+ * @property {string} id
+ * @property {string} name
+ * @property {string} webContentLink - link to download shared file
+ * @property {string} webViewLink - link to view shared file (in Google App, or browser)
+ */
+
+/**
+ * Returns id of first folder found by provided name
+ * @param {string} folderName
+ * @returns {Promise.<string>}
+ */
+function getFolderId(folderName) {
+  return new Promise((resolve, reject) => {
+    drive.files.list({
+      q: `name='${folderName}' and mimeType = 'application/vnd.google-apps.folder'`,
+      fields: 'nextPageToken, files(id, name)',
+    }, (err, response) => {
+      if (err) {
+        reject(err);
+        return;
+      }
+
+      if (!response.files || !response.files.length) {
+        const msg = `You must have ${folderName} directory in the root of your google drive`;
+        reject(new Error(msg));
+        return;
+      }
+
+      resolve(response.files[0].id);
+    });
+  });
+}
+
+/**
+ * Uploads stream to google drive
+ * @param {string} parentDirectoryId - you must specify directory for files
+ * @param {stream.Readable} stream
+ * @param {string} stream.filename
+ * @param {function(GoogleDriveFileResource)} resolve
+ * @param {function(Error)} reject
+ */
+function uploadToGoogleDrive(parentDirectoryId, stream, resolve, reject) {
+  drive.files.create({
+    resource: {
+      name: stream.filename,
+      parents: [parentDirectoryId],
+    },
+    media: {
+      body: stream,
+    },
+    fields: 'id,name,webContentLink,webViewLink',
+  }, (err, result) => {
+    if (err) {
+      reject(err);
+      return;
+    }
+
+    resolve(result);
+  });
+}
+
+/**
+ * @param {http.Request} req
+ * @param {string} fieldName - where to look for file
+ * @param {object} options
+ * @param {string} options.folder - directory to upload to
+ * @returns {Promise.<GoogleDriveFileResource, Error>}
+ */
+export async function upload(req, fieldName, options) {
+  const parentDirectoryId = await getFolderId(options.folder);
```

Comment:
Then it may be easier to call 'setup' instead of 'ensureGoogleDriveInitialized'? )